src/bitcoind/utils.rs

```diff
@@ -288,3 +294,93 @@ pub fn emer_txid(
 
     Ok(None)
 }
+
+/// CPFP a bunch of transactions, bumping their feerate by at least `target_feerate`.
+/// `target_feerate` is expressed in sat/kWU.
+/// All the transactions' feerate MUST be below `target_feerate`.
+pub fn cpfp_package(
+    revaultd: &Arc<RwLock<RevaultD>>,
+    bitcoind: &BitcoinD,
+    to_be_cpfped: Vec<ToBeCpfped>,
+    target_feerate: u64,
+) -> Result<Vec<Txid>, BitcoindError> {
+    let revaultd = revaultd.read().unwrap();
+    let cpfp_descriptor = &revaultd.cpfp_descriptor;
+    // First of all, compute all the information we need from the to-be-cpfped transactions.
+    let mut txids = HashSet::with_capacity(to_be_cpfped.len());
+    let mut package_weight = 0;
+    let mut package_fees = Amount::from_sat(0);
+    let mut txins = Vec::with_capacity(to_be_cpfped.len());
+    for tx in to_be_cpfped.iter() {
+        txids.insert(tx.txid());
+        package_weight += tx.max_weight();
+        package_fees += tx.fees();
+        assert!(((package_fees.as_sat() * 1000 / package_weight) as u64) < target_feerate);
+        match tx.cpfp_txin(cpfp_descriptor, &revaultd.secp_ctx) {
+            Some(txin) => txins.push(txin),
+            None => {
+                log::error!("No CPFP txin for tx '{}'", tx.txid());
+                return Ok(txids.into_iter().collect());
+            }
+        }
+    }
```

Comment:
Rather than returning all the transactions, I need to throw an error here. I'm not sure which one

Reply:
You can create a new enum variant for it if none other fit.